src/io/rgb_led.c

```diff
@@ -74,6 +75,131 @@ void rgb_ledflash(int color1, int color2, uint32_t period, int duty) {
   }
 }
 
+// speed of movement
+float RAINBOW_SPEED = 1.0f;
```

Comment:
we need to find a better solution for these patterns long term. just coding them out one-by-on is not gonna be scalable.

Reply:
I agree we cannot code out infinite patterns one by one, but even providing a couple of simple patterns to select from will make it much more user-friendly than 'other' firmwares. 
Single color (settable via configurator)
-> solid color
-> flash single color
-> Knight rider
Dual color (settable via configurator)
-> flash/fade alternate colors
-> fade based on thottle
-> fade based on battery
-> chasing colors?

Rainbow (because i like this one:))

Reply:
Well that didnt format nicely...

Reply:
notice how the number of combinations _just off your personal preferences_ explodes....no, sorry, i think we have to better than that.
in an ideal world you would have a fixed set of transitions for a single led and a means of chaining multiple of these together. this sequence is programmed via the configurator, which can contain presets for some patterns.
make this data vs code driven.

Reply:
Sure data driven will make it far more extensible... Just think that we dont want to make the users learn a new (led coding) language to implement some funky color changes. I've tried to set up LEDs on Betaflight and its pretty confusing...
Hence my argument for just having a few set patterns (there should be only 4 patterns in the prev message: 1 flashing; 2 with modifier (channel,battery), 3 knight rider/chaser, and 4 rainbow).
However if there is a way to make this data driven then i suppose all those preset patterns could be provided as samples

Reply:
I pushed a commit last night that allows for up to 50 sequence steps, each step allowing you to specify up to 3 colors (or off) for each of your leds (up to the 32 led max we discussed). 
Wave sequence is a separate customizable pattern, currently used as a rainbow, but the user might want pulsing waves of a single or two colors, which is now possible (would require a lot more data to achieve that in the above sequence step feature.
A third option is for a solid colour, with an optional modifier channel to allow you to flip colors on a switch, or to have it controlled by throttle for a cool effect.
Low battery flash colors are customizable, as is the failsafe colors (you may want to select colors that are easier to see from a distance).
I think this should cover us for all user requests, and I think its a lot more customizable than *flight firmwares :)